### PR TITLE
HMisOver

### PR DESCRIPTION
**Pull request by Yam0r**

### CODE REVIEW

**Review by okuzan — CHANGES_REQUESTED** [review]
Please fix the build

File: src/main/java/core/basesyntax/Circle.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import java.text.DecimalFormat;
+import java.util.Random;
+
+public class Circle extends ColorSupplier {
+    private int radius;
+    private String color;
+
+    public Circle() {
+        this.radius = new Random().nextInt(4) + 1;
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/Circle.java:11]: no need to create a random value right here, you need to pass it from outside to the constructor

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class ColorSupplier implements Draw, Figure {
+    public static String getRandomColor() {
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/ColorSupplier.java:6]:
```suggestion
    public String getRandomColor() {
```

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class ColorSupplier implements Draw, Figure {
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/ColorSupplier.java:5]:
```suggestion
public class ColorSupplier {
```

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    @Override
+    public void draw() {
+
+    }
+
+    @Override
+    public double getArea() {
+        return 0;
+    }
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/ColorSupplier.java:20]:
```suggestion
```

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+    public enum Color {
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/ColorSupplier.java:22]: Move to the separate file enum Color

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        Green,
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/ColorSupplier.java:23]: all variables in enam are written in the upper case

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 7]
```diff
@@ -0,0 +1,5 @@
+package core.basesyntax;
+
+public class FigureSupplier {
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/FigureSupplier.java:3]: image

File: src/main/java/core/basesyntax/Circle.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+import java.text.DecimalFormat;
+import java.util.Random;
+
+public class Circle extends ColorSupplier {
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/Circle.java:6]:
```suggestion
public class Circle extends Figure {
```

File: src/main/java/core/basesyntax/Figure.java [hunk 9]
```diff
@@ -0,0 +1,5 @@
+package core.basesyntax;
+
+public interface Figure {
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/Figure.java:3]:
```suggestion
public interface Figure implements Drawable, AreaCalculator {
```

File: src/main/java/core/basesyntax/Figure.java [hunk 10]
```diff
@@ -0,0 +1,5 @@
...
+    double getArea();
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/Figure.java:4]: move this method to the separate interface AreaCalculator

File: src/main/java/core/basesyntax/Draw.java [hunk 11]
```diff
@@ -0,0 +1,5 @@
+package core.basesyntax;
+
+public interface Draw {
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/Draw.java:3]:
```suggestion
public interface Drawable {
```

File: src/main/java/core/basesyntax/IsoscelesTrapezoid.java [hunk 12]
```diff
@@ -0,0 +1,31 @@
+package core.basesyntax;
+
+import java.text.DecimalFormat;
+import java.util.Random;
+
+public class IsoscelesTrapezoid extends ColorSupplier {
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/IsoscelesTrapezoid.java:6]: same as other figures

File: src/main/java/core/basesyntax/Rectangle.java [hunk 13]
```diff
@@ -0,0 +1,29 @@
+package core.basesyntax;
+
+import java.text.DecimalFormat;
+import java.util.Random;
+
+public class Rectangle extends ColorSupplier {
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/Rectangle.java:6]: same as other figures

File: src/main/java/core/basesyntax/RightTriangle.java [hunk 14]
```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+import java.text.DecimalFormat;
+
+public class RightTriangle extends ColorSupplier {
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/RightTriangle.java:5]: same as other figures

File: src/main/java/core/basesyntax/Square.java [hunk 15]
```diff
@@ -0,0 +1,25 @@
+package core.basesyntax;
+
+import java.text.DecimalFormat;
+
+public class Square extends ColorSupplier {
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/Square.java:5]: same as other figures

**Review by kozhukhovsky — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+        //for the generation of figures with episodic powers
+        int randvalues = new Random().nextInt(4) + 1;
+        return randvalues;
+    }
+
+    public String getDefaultColor() {
+        String color;
+        color = "white";
+        return color;
+    }
```
Comment (olekskov) [inline src/main/java/core/basesyntax/FigureSupplier.java:23]: Methods and signature of your class not comply with it's purpose
There should be something like below
```suggestion
public class FigureSupplier {
    private final Random random = new Random();
    private final ColorSupplier colorSupplier = new ColorSupplier();

    public Figure getRandomFigure() {
        ...some code here
    }

    public Figure getDefaultFigure() {
        ...some code
    }
```

File: src/main/java/core/basesyntax/Main.java [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
+        figures[0] = new Circle();
+        figures[1] = new IsoscelesTrapezoid();
+        figures[2] = new Rectangle();
+        figures[3] = new RightTriangle(3, 4, "Green");
+        figures[4] = new Square(4, "Green");
+
+        // Выводим все фигуры
+        for (Figure figure : figures) {
+            figure.draw();
+        }
```
Comment (olekskov) [inline src/main/java/core/basesyntax/Main.java:15]:
```suggestion
        FigureSupplier figureSupplier = new FigureSupplier();
        Figure[] figures = new Figure[FIGURES_AMOUNT];
        
        for (int i = 0; i < FIGURES_AMOUNT; i++) {
            ...some logic to fill array using figureSupplier.getRandomFigure() and figureSupplier.getDefaultFigure() methods
        }
```

**Review by olekskov — COMMENTED** [review]
Please check new comments to redesign FigureSupplier and also check comments from previous review. Most of them are not fixed

**Review by olekskov — CHANGES_REQUESTED** [review]
`

**Pull request closed without merging**